### ISSUE TITLE
Iterative fitting with sigma clipping

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#4698]
Is it possible to use `modeling` and its fitters for such a workflow?

1. Create a model.
2. Attach a fitter to the model.
3. Fit with some data.
4. Reject some outliers with sigma clipping.
5. Redo fitting with clipped data.
6. Repeat 4-5 for a number of times (`niter`) until satisfied.

Using example from http://docs.astropy.org/en/latest/modeling/index.html#simple-1-d-model-fitting , maybe something like this?

```python
g = fit_g(g_init, x, y, do_sigma_clip=True, sigma=3.0, niter=10)
```

c/c @hcferguson , @larrybradley , @ibusko

**Comment by crawfordsm** [comment]
I'd definitely like to see this in astropy and have written wrapper code to do something similar, but having it built into the fitter functions directly would be very useful.

**Comment by nocturnalastro** [comment]
For optimizers which allow callbacks this would be really easy.

**Comment by mirca** [comment]
I will submit an WIP pull request for this soon :-).

**Comment by mirca** [comment]
@pllim I think this can be closed since #4760 is merged.

**Comment by pllim** [comment]
Okay, thanks!
